src/main/java/org/jenkinsci/plugins/workflow/cps/SnippetizerLink.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.workflow.cps;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.Functions;
+import hudson.model.RootAction;
+
+import javax.annotation.Nonnull;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.jenkinsci.plugins.workflow.cps.Snippetizer.ACTION_URL;
+
+/**
+ * A link that will show up on the side panel of the snippet generator and other similar pages.
+ * Display order is determined by extension ordinal - highest ordinal first.
+ * 
+ * @author Andrew Bayer
+ */
+public abstract class SnippetizerLink implements ExtensionPoint {
+    private static final Logger LOGGER = Logger.getLogger(SnippetizerLink.class.getName());
+
+    /**
+     * Get the URL this link should point to.
+     */
+    @Nonnull
+    public abstract String getUrl();
+
+    /**
+     * Get the actual URL to use in sidepanel.jelly. If {@link #getUrl()} is not absolute, this will prepend "../" to
```

Comment:
Why would you not just require `getUrl` to be absolute?

Reply:
Good question. Will change to absolute.